resizable-rect/demo.js

```diff
@@ -0,0 +1,278 @@
+/**
+ * Adds resizable geo rect to map
+ *
+ * @param {H.Map} map                      A HERE Map instance within the application
+ * @param {H.mapevents.Behavior} behavior  Behavior implements default interactions with map
+ */
+function createResizableRect(map, behavior) {
+  var rect =  new H.map.Rect(
+        new H.geo.Rect(51.2, 20.5, 49.5, 24.5),
+        {
+          style: {fillColor: 'rgba(100, 100, 100, 0.5)', lineWidth: 0}
+        }
+      ),
+      rectOutline = new H.map.Polyline(
+        rect.getGeometry().getExterior(),
+        {
+          style: {lineWidth: 8, strokeColor: 'rgba(255, 0, 0, 0)', fillColor: 'rgba(0, 0, 0, 0)', lineCap: 'square'}
+        }
+      ),
+      rectGroup = new H.map.Group({
+        volatility: true, // mark the group as volatile for smooth dragging of all it's objects
+        objects: [rect, rectOutline]
+      }),
+      rectTimeout;
+
+  // ensure that the objects can receive drag events
+  rect.draggable = true;
+  rectOutline.draggable = true;
+
+  // extract first point of the rect's outline polyline's LineString and
+  // push it to the end, so the outline has a closed geometry
+  rectOutline.getGeometry().pushPoint(rectOutline.getGeometry().extractPoint(0));
+
+  // add group with rect and it's outline (polyline)
+  map.addObject(rectGroup);
+
+  // event listener for rect group to show outline (polyline) if moved in with mouse (or touched on touch devices)
+  rectGroup.addEventListener('pointerenter', function(evt) {
+    var group = this,
+        rectOutline = group.getObjects()[1],
```

Comment:
Like in the other examples I think it's better to use the outer scope variable.

Reply:
done